src/AppShed/Extensions/SpreadsheetMapsBundle/Controller/MapsController.php

```diff
@@ -89,48 +89,53 @@ public function documentAction(Request $request)
             return (new Remote($screen))->getSymfonyResponse();
         }
         $address = $doc->getAddress();
+        if (!$address) {
+            $address = 'Address';
+        }
+
         try {
+
             $document = $this->getDocument(
                 $doc->getKey(),
                 $this->getFilterString($doc->getFilters(), $request)
             );
-            //This screen will have a list of the values in A column
-            $screen = new Screen($document->getTitle());
+
+            //This screen will have a list of the markers in Address column
+            $screen = new Map($document->getTitle());
+
             //For each row of the table
             foreach ($document as $entry) {
                 $index = true;
                 $lines = $entry->getCustom();
+
+
                 //Each of the columns of the row
                 foreach ($lines as $customEntry) {
+
                     $name = $customEntry->getColumnName();
                     $value = $customEntry->getText();
+
                     //If the name of a column ends with a '-' then we dont show it
                     if (((strlen($name) - 1) == strpos($name, '-')) == false) {
                         if ($index == true) {
                             //This screen will have all the values across the row
                             $innerScreen = new Screen($value);
-                            $link = new Link($value);
-                            $screen->addChild($link);
                             $index = false;
-                            $link->setScreenLink($innerScreen);
                         } else {
                             if (!empty($value)) {
-                                $map = false;
-                                if ($name == $address ) {
+                                if ($name == strtolower($address) ) {
```

Comment:
why are you not settings `$address` to `strtolower($doc->getAddress())` at the beginning, so you don't do it in the loop.

Also, what if `$name` is not all lowercase?